pkg/build/controller/buildpod/controller.go

```diff
@@ -236,7 +246,7 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 		case buildapi.BuildPhaseError, buildapi.BuildPhaseFailed:
 			bc.recorder.Eventf(build, kapi.EventTypeNormal, buildapi.BuildFailedEventReason, fmt.Sprintf(buildapi.BuildFailedEventMessage, build.Namespace, build.Name))
 		}
-		common.HandleBuildCompletion(build, bc.runPolicies)
+		common.HandleBuildCompletion(build, bc.buildLister, bc.buildConfigGetter, bc.buildDeleter, bc.runPolicies)
```

Comment:
if you are just setting the limits on a BC, what would trigger the pruning then?  saving the BC?